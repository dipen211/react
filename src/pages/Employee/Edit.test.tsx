import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mountToJson } from 'enzyme-to-json';
import EditEmployee from './Edit';
import { RouteComponentProps } from 'react-router-dom';
import axios from 'axios';
Enzyme.configure({ adapter: new Adapter() });
const mockfn = jest.fn();

jest.mock('axios', () => {
    const exampleArticles = [
        { title: 'employee', url: 'http://localhost:5000/employees/' }
    ];

    return {
        get: jest.fn(() => Promise.resolve(exampleArticles)),
    };
});

describe('EditEmployee', () => {
    let wrapper: any;

    const props: RouteComponentProps<{}> = {
        history: mockfn,
        location: mockfn,
        match: mockfn,
    }
    beforeEach(() => {
        wrapper = shallow(<EditEmployee {...props} />);
    });
    it('should match the snapshot', () => {
        expect(mountToJson(wrapper)).toMatchSnapshot();
    });

    it('Should capture firstname correctly onChange', () => {
        wrapper.find('Field').first();
        wrapper.simulate('change', { preventDefault: () => { } });
        wrapper.setState({ first_name: "Dipen" })
        expect(wrapper.state('first_name')).toEqual('Dipen');
    })
    it('Should capture last_name correctly onChange', () => {
        wrapper.find('Field').at(1);
        wrapper.simulate('change', { preventDefault: () => { } });
        wrapper.setState({ last_name: "Dipen" })
        expect(wrapper.state('last_name')).toEqual('Dipen');
    })
    it('Should capture email correctly onChange', () => {
        wrapper.find('input').at(2);
        wrapper.simulate('change', { preventDefault: () => { } });
        wrapper.setState({ email: "Dipen" })
        expect(wrapper.state('email')).toEqual('Dipen');
    });

    it('Should capture password correctly onChange', () => {
        wrapper.find('input').at(3);
        wrapper.simulate('change', { preventDefault: () => { } });
        wrapper.setState({ password: "Dipen" })
        expect(wrapper.state.employee('password')).toEqual('Dipen');
    });
    it('handle handleInputChanges', () => {
        wrapper.instance().handleInputChanges({ target: { value: 'test', name: 'email' } });
        wrapper.update();
        expect(wrapper.state('email')).toEqual('test');
    });
    it('handle setValues', () => {
        wrapper.instance().setValues({ target: { value: 'test', name: 'email' } });
        wrapper.update();
        expect(wrapper.state('email')).toEqual('test');
    });
    it('should submit a valid form', () => {
        const state = {
            id: "1", team_id: "1",
            first_name: 'Dipen', last_name: 'Dipen',
            email: 'Dipen@gmail.com',
            password: "dipen",
        };
        const expectedArg = "id: 1, first_name: Dipen, last_name: Dipen, email: Dipen@gmail.com, password: dipen";
        window.alert = jest.fn();
        wrapper.setState(state);
        wrapper.find('Formik').simulate('submit', { preventDefault: () => { } });
        expect(window.alert).toHaveBeenCalledWith(expectedArg);
    });
})