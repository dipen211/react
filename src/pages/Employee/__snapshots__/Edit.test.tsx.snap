// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditEmployee should match the snapshot 1`] = `
<Fragment>
  <div
    className="col-md-12 form-wrapper"
  >
    <h1>
       
      Edit Employee
       
    </h1>
    <div
      className="alert alert-info"
      role="alert"
    >
      Fill the form below to edit Employee
    </div>
    <Formik
      initialValues={Object {}}
      onSubmit={[Function]}
      validationSchema={
        ObjectSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_defaultDefault": [Function],
          "_deps": Array [],
          "_excludedEdges": Array [],
          "_exclusive": Object {},
          "_mutate": undefined,
          "_nodes": Array [
            "password",
            "email",
            "last_name",
            "first_name",
            "id",
          ],
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_sortErrors": [Function],
          "_type": "object",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "fields": Object {
            "email": StringSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "email": false,
                "required": true,
              },
              "_mutate": undefined,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
              "type": "string",
            },
            "first_name": StringSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "required": true,
              },
              "_mutate": undefined,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
              "type": "string",
            },
            "id": StringSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "required": true,
              },
              "_mutate": undefined,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
              "type": "string",
            },
            "last_name": StringSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "required": true,
              },
              "_mutate": undefined,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
              "type": "string",
            },
            "password": StringSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "min": true,
                "required": true,
              },
              "_mutate": undefined,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
              "type": "string",
            },
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
          "type": "object",
        }
      }
    >
      <Component />
    </Formik>
  </div>
</Fragment>
`;
